lib/AST/ASTImporter.cpp

```diff
@@ -2286,6 +2333,16 @@ Decl *ASTNodeImporter::VisitRecordDecl(RecordDecl *D) {
       // We're going to have to compare D against potentially conflicting Decls, so complete it.
       if (D->hasExternalLexicalStorage() && !D->isCompleteDefinition())
         D->getASTContext().getExternalSource()->CompleteType(D);
+    } else {
+      // DeclContext based lookup may not find forward declared structs in C.
+      // For instance, consider that this code had been already parsed into the
+      // "To" context:
+      //   struct A { struct X *Xp; };
+      // In this case localUncachedLookup will not find the forward decl of X.
+      // Thus, we do an identifier based search through the front end specific
+      // void pointers whic are injected into the DeclarationName objects.
+      if (!Importer.getToContext().getLangOpts().CPlusPlus)
```

Comment:
Probably check for something like "C" instead of "not C++" (that can be something other like ObjC?).